File: grida-apis/end-user-app/src/main/java/org/grida/user/controller/UserController.java
package org.grida.user.controller;

import io.wwan13.wintersecurity.resolve.RequestUserSubject;
import lombok.RequiredArgsConstructor;
import org.grida.dto.IdResponse;
import org.grida.dto.UserEmailResponse;
import org.grida.response.ApiResponse;
import org.grida.user.dto.request.ChangeActivateImageRequest;
import org.grida.user.dto.request.ProfileImageRequest;
import org.grida.user.dto.response.GetActivateImageResponse;
import org.grida.user.dto.response.GetProfileImageHistoryResponse;
import org.grida.user.usecase.*;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping(value = "/api/user")
@RequiredArgsConstructor
public class UserController {

    private final OnboardingUseCase onboardingUseCase;
    private final GetActivateImageUseCase getActivateImageUseCase;
    private final RefreshProfileImageUseCase refreshProfileImageUseCase;
    private final GetProfileImageHistoryUseCase getProfileImageHistoryUseCase;
    private final ChangeActivateImageUseCase changeActivateImageUseCase;

    @PostMapping("/onboarding")
    public ApiResponse onboarding(
            @RequestUserSubject String userEmail,
            @RequestBody ProfileImageRequest request
    ) {
        UserEmailResponse response = onboardingUseCase.execute(userEmail, request);
        return ApiResponse.ok(response);
    }

    @GetMapping("/image")
    public ApiResponse getActivateImage(@RequestUserSubject String userEmail) {
        GetActivateImageResponse response = getActivateImageUseCase.execute(userEmail);
        return ApiResponse.ok(response);
    }

    @PostMapping("/image/refresh")
    public ApiResponse refreshProfileImage(
            @RequestUserSubject String userEmail,
            @RequestBody ProfileImageRequest request
    ) {
        UserEmailResponse response = refreshProfileImageUseCase.execute(userEmail, request);
        return ApiResponse.ok(response);
    }

    @GetMapping("/image/history")
    public ApiResponse getProfileImageHistory(@RequestUserSubject String userEmail) {
        GetProfileImageHistoryResponse response = getProfileImageHistoryUseCase.execute(userEmail);
        return ApiResponse.ok(response);
    }

    @PatchMapping("/image/change")
    public ApiResponse changeActivateProfileImage(
            @RequestUserSubject String userEmail,
            @RequestBody ChangeActivateImageRequest request
    ) {
        IdResponse response = changeActivateImageUseCase.execute(userEmail, request);
        return ApiResponse.ok(response);
    }
}

File: grida-core/core-api/src/main/kotlin/org/grida/presentation/v1/user/UserController.kt
package org.grida.presentation.v1.user

import org.grida.api.ApiResponse
import org.grida.api.dto.IdResponse
import org.grida.application.user.UpdateProfileRequest
import org.grida.application.user.UpdateProfileUseCase
import org.grida.application.user.UpdateThemeRequest
import org.grida.application.user.UpdateThemeUseCase
import org.grida.support.RequestUserId
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/v1/user")
class UserController(
    private val updateProfileUseCase: UpdateProfileUseCase,
    private val updateThemeUseCase: UpdateThemeUseCase
) {

    @PostMapping("/profile")
    fun updateProfile(
        @RequestUserId userId: Long,
        @RequestBody request: UpdateProfileRequest
    ): ApiResponse<IdResponse> {
        updateProfileUseCase.execute(userId, request)
        val response = IdResponse(userId)

        return ApiResponse.success(response)
    }

    @PostMapping("/theme")
    fun updateTheme(
        @RequestUserId userId: Long,
        @RequestBody request: UpdateThemeRequest
    ): ApiResponse<IdResponse> {
        updateThemeUseCase.execute(userId, request)
        val response = IdResponse(userId)

        return ApiResponse.success(response)
    }
}
